servers/rendering/renderer_rd/renderer_scene_sky_rd.cpp

```diff
@@ -398,55 +402,139 @@ void RendererSceneSkyRD::ReflectionData::update_reflection_data(int p_size, int
 			mm.size.height = mmh;
 			mm.view = RD::get_singleton()->texture_create_shared_from_slice(RD::TextureView(), downsampled_radiance_cubemap, 0, j, RD::TEXTURE_SLICE_CUBEMAP);
 
+			if (prefer_raster_effects) {
+				// we need a framebuffer for each side of our cubemap
+
+				for (int k = 0; k < 6; k++) {
+					mm.views[k] = RD::get_singleton()->texture_create_shared_from_slice(RD::TextureView(), downsampled_radiance_cubemap, k, j);
+					Vector<RID> fbtex;
+					fbtex.push_back(mm.views[k]);
+					mm.framebuffers[k] = RD::get_singleton()->framebuffer_create(fbtex);
+				}
+			}
+
 			mmw = MAX(1, mmw >> 1);
 			mmh = MAX(1, mmh >> 1);
 		}
 	}
 }
 
 void RendererSceneSkyRD::ReflectionData::create_reflection_fast_filter(RendererStorageRD *p_storage, bool p_use_arrays) {
-	p_storage->get_effects()->cubemap_downsample(radiance_base_cubemap, downsampled_layer.mipmaps[0].view, downsampled_layer.mipmaps[0].size);
+	EffectsRD *effects = p_storage->get_effects();
+	ERR_FAIL_NULL_MSG(effects, "Effects haven't been initialised");
+	bool prefer_raster_effects = effects->get_prefer_raster_effects();
 
-	for (int i = 1; i < downsampled_layer.mipmaps.size(); i++) {
-		p_storage->get_effects()->cubemap_downsample(downsampled_layer.mipmaps[i - 1].view, downsampled_layer.mipmaps[i].view, downsampled_layer.mipmaps[i].size);
-	}
+	if (prefer_raster_effects) {
+		for (int k = 0; k < 6; k++) {
+			effects->cubemap_downsample_raster(radiance_base_cubemap, downsampled_layer.mipmaps[0].framebuffers[k], k, downsampled_layer.mipmaps[0].size);
+		}
 
-	Vector<RID> views;
-	if (p_use_arrays) {
-		for (int i = 1; i < layers.size(); i++) {
-			views.push_back(layers[i].views[0]);
+		for (int i = 1; i < downsampled_layer.mipmaps.size(); i++) {
+			for (int k = 0; k < 6; k++) {
+				effects->cubemap_downsample_raster(downsampled_layer.mipmaps[i - 1].view, downsampled_layer.mipmaps[i].framebuffers[k], k, downsampled_layer.mipmaps[i].size);
+			}
+		}
+
+		if (p_use_arrays) {
+			for (int i = 0; i < layers.size(); i++) {
```

Comment:
The code here does not work properly, this needs to be solved differently or I'm missing something in the fragment shader that allows it to behave the same as its compute counterpart. 

Anyway, with realtime updating the sky, roughness set on our material and using reflection arrays we get this ugly result:
![image](https://user-images.githubusercontent.com/1945449/127418570-ab6276d1-2473-40c1-bbb9-12d97cd4f881.png)

It's supposed to look something like this:
![image](https://user-images.githubusercontent.com/1945449/127418605-2c06cd72-91cb-4d34-83c3-dbfc6201d04b.png)

Reply:
It actually doesn't look great with reflection arrays turned off either:
![image](https://user-images.githubusercontent.com/1945449/127418818-8f25a796-cf1b-44cd-a131-62d8053f04bb.png)

So its definitely something I'm not implementing right in the downsample shader.

Reply:
This is the sky shader I'm using to test this:
```
shader_type sky;

uniform sampler2D source_panorama : filter_linear;

void sky() {
	vec2 uv = SKY_COORDS;
	uv.x += TIME * 0.01;
	COLOR = texture(source_panorama, uv).rgb;
}
```

Reply:
Final reference, this is what it looks like using the clustered shader:
![image](https://user-images.githubusercontent.com/1945449/127419488-70ea77dd-2f12-4f12-9dad-239edec13a51.png)